addons/point_of_sale/models/pos_order.py

```diff
@@ -139,7 +139,6 @@ def _process_order(self, order, draft, existing_order):
 
         if pos_order.to_invoice and pos_order.state == 'paid':
             pos_order.action_pos_order_invoice()
-            pos_order.account_move.sudo().with_context(force_company=self.env.user.company_id.id).post()
```

Comment:
Note that if no new move is created (we go through [1]), the `account_move` is not posted. However, I never went through [1] during my tests, so not sure if it is still useful.

[1] https://github.com/odoo/odoo/pull/41525/files#diff-5eeb7d6a1a488dff039e5cc3300730e1R359

Reply:
I think it should never happens to have an invoice created from the pos, and not posted.